### PR TITLE
applications: nrf5340_audio: Revert to adv after short burst of dir_adv

### PR DESCRIPTION
**Pull request by alexsven**
- If already bonded, do a short burst of directed advertising before reverting back to a slower pace advertising
- OCT-2618

### CODE REVIEW

**Comment by alexsven** [comment]
Currently blocked by https://github.com/zephyrproject-rtos/zephyr/issues/68681

**Review by rick1082 — COMMENTED** [review]
Would we need to update the unicast client too? Since the unicast client might only search for direct ADV after bonded with 2 headsets.

**Review by rick1082 — CHANGES_REQUESTED** [review]
Maybe we can have a config for enabling this feature, which dedicated for headset to work with phone. Otherwise we need to modify our gateway for being able to scan and connect device which using normal ADV, since there could be a chance gateway missed dir-ADV.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -267,7 +267,8 @@ nRF5340 Audio
 * Updated:
 
   * ISO data sending has been refactored, and is now done in a single file: bt_le_audio_tx.
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:269]: This and other edits in this section are addressing the changelog formatting, where "Updated" needs to match the sentences in the bullets.
```suggestion
  * Sending of the ISO data, which is now done in a single file :file:`bt_le_audio_tx`.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by alexsven** [comment]
> Would we need to update the unicast client too? Since the unicast client might only search for direct ADV after bonded with 2 headsets.

Nice catch, fixed and tested

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (alexsven) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:269]: This is strictly not part of this PR, same for the comment below

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -267,7 +267,8 @@ nRF5340 Audio
...
-  * Split the generic applications into their own samples with separate main.c files
+  * Split the generic applications into their own samples with separate main.c files.
+  * Revert back to slower advertising after a short burst (1.28s) of directed advertising.
```
[the remark this answers is not in the document]
Comment (alexsven) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:271]: That sentence doesn't make sense. Is the thought that the full sentence is: "Updated advertising process by revert...."?

File: applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h [hunk 3]
```diff
@@ -14,4 +14,12 @@
  */
 void bt_mgmt_adv_init(void);
 
+/**
+ * @brief	Handle timed out directed advertisement.
```
[the remark this answers is not in the document]
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h:18]: I don't agree: https://www.oxfordlearnersdictionaries.com/definition/american_english/time-out

File: applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h [hunk 4]
```diff
@@ -14,4 +14,12 @@
...
+ *
+ * @note	Will delete the old ext_adv and create a new one,
+ *		as well as setting the dir_adv_timed_out flag and restarting advertisement.
```
[the remark this answers is not in the document]
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h:21]: Sure, will remove `@note` and then make the changes outlined here

File: applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c [hunk 5]
```diff
@@ -79,6 +79,19 @@ static void connected_cb(struct bt_conn *conn, uint8_t err)
 	uint8_t num_conn = 0;
 	struct bt_mgmt_msg msg;
 
+	if (err == BT_HCI_ERR_ADV_TIMEOUT && IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c:82]: So, timed out ADVs come back through the connected_cb?

**Review by koffes — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c (hunk 5 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c:82]: Correct

**Review by rick1082 — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Yes, as I wrote above "This and other edits in this section are addressing the changelog formatting, where "Updated" needs to match the sentences in the bullets".

File: applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (greg-fer) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h:18]: You are linking to an entry about verb. I'm editing a compound modifier, which requires a hyphen.

[the remark this answers is not in the document]
Comment (greg-fer) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv_internal.h:18]: You are linking to an entry about verb. I'm editing a compound modifier, which requires a hyphen.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:269]: OK, I'll open a separate PR...

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -267,7 +267,8 @@ nRF5340 Audio
 * Updated:
 
   * ISO data sending has been refactored, and is now done in a single file: bt_le_audio_tx.
-  * Split the generic applications into their own samples with separate main.c files
+  * Split the generic applications into their own samples with separate main.c files.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:270]: OK, I'll open a separate PR...

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexsven) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Ok, will change. Is that a new thing? I looked through older release notes, and didn't see that any of them were written like that.

**Comment by alexsven** [comment]
> Maybe we can have a config for enabling this feature, which dedicated for headset to work with phone. Otherwise we need to modify our gateway for being able to scan and connect device which using normal ADV, since there could be a chance gateway missed dir-ADV.

No need to have a config. The gateway will accept either directed adv or normal advertising as long as it is from a known peripheral

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (alexsven) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Don't need to, I've changed them here now.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (alexsven) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Don't need to, I've changed them here now.

**Review by greg-fer — APPROVED** [review]

**Comment by alexsven** [comment]
> > > Maybe we can have a config for enabling this feature, which dedicated for headset to work with phone. Otherwise we need to modify our gateway for being able to scan and connect device which using normal ADV, since there could be a chance gateway missed dir-ADV.
> > 
> > 
> > No need to have a config. The gateway will accept either directed adv or normal advertising as long as it is from a known peripheral
> 
> Then we have to setup the unique naming for any occasions if there are more than two demo sets : P Since now bonded headsets can be found by any gateways.

Good point, will look into Filter Accept List

**Review by FrancescoSer — APPROVED** [review]

**Review by gWacey — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c [hunk 7]
```diff
@@ -72,6 +72,19 @@ static void bond_find(const struct bt_bond_info *info, void *user_data)
 	}
 }
 
+static void filter_accept_list_add(const struct bt_bond_info *info, void *user_data)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_BT_FILTER_ACCEPT_LIST)) {
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c:79]: This is only called when CONFIG_BT_FILTER_ACCEPT_LIST is enabled. I'd argue we either remove this check or return an error if it is called when this flag is not set.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 8]
```diff
@@ -254,6 +254,8 @@ Serial LTE modem
 nRF5340 Audio
 -------------
 
+* Added support for Filter Accept List.
```
Comment (koffes) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:257]: Suggest we add some more info. E.g. that this is enabled by default and how it influences the user experience.

**Review by koffes — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c (hunk 7 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c:79]: It is to guard bt_le_filter_accept_list_add from the linker if FAL is not enabled, but I'll test to see if it's needed after I moved the code below

Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c]: Removed

**Review by koffes — APPROVED** [review]

**Pull request merged**